### PR TITLE
Add `latestMilestone`, `confirmedMilestone`, and `baseToken` informations to InfoResponse

### PR DESCRIPTION
**Pull request by samuel-rufi**

### CODE REVIEW

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 1]
```diff
@@ -54,8 +74,22 @@ pub struct ProtocolResponse {
...
+/// Returned in [`InfoResponse`].
+/// Rent information about the node.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct BaseTokenResponse {
+    pub name: String,
+    #[serde(rename = "tickerSymbol")]
+    pub ticker_symbol: String,
+    pub unit: String,
+    pub decimals: u8,
+    pub subunit: String,
```
Comment (Thoralf-M) [inline bee-api/bee-rest-api/src/types/responses.rs]: Maybe change the order?
```suggestion
    pub unit: String,
    pub subunit: String,
    pub decimals: u8,
```

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 2]
```diff
@@ -54,8 +74,22 @@ pub struct ProtocolResponse {
     pub min_pow_score: f64,
     #[serde(rename = "rentStructure")]
     pub rent_structure: RentStructureResponse,
-    #[serde(rename = "tokenSupply")]
-    pub token_supply: String,
+    #[serde(rename = "baseToken")]
+    pub base_token: BaseTokenResponse,
```
Comment (alexsporn) [inline bee-api/bee-rest-api/src/types/responses.rs]: This should be moved to the InfoResponse

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 3]
```diff
@@ -54,8 +74,22 @@ pub struct ProtocolResponse {
     pub min_pow_score: f64,
     #[serde(rename = "rentStructure")]
     pub rent_structure: RentStructureResponse,
-    #[serde(rename = "tokenSupply")]
```
Comment (alexsporn) [inline bee-api/bee-rest-api/src/types/responses.rs]: re-introduce the tokenSupply

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 4]
```diff
@@ -30,22 +30,42 @@ impl BodyInner for InfoResponse {}
...
+    pub timestamp: u32,
+    #[serde(rename = "milestoneId")]
+    pub milestone_id: String,
+}
+
 /// Returned in [`InfoResponse`].
 /// Protocol information about the node.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct ProtocolResponse {
-    #[serde(rename = "protocolVersion")]
```
Comment (alexsporn) [inline bee-api/bee-rest-api/src/types/responses.rs:47]: do not remove the protocol version, rename it to `version`

File: bee-api/bee-rest-api/src/types/responses.rs (hunk 1 above)
Comment (alexsporn) [inline bee-api/bee-rest-api/src/types/responses.rs]: subunit is optional and should be ommited if empty (not sure if this is the default behaviour of serde)

**Review by Thoralf-M — DISMISSED** [review]

File: bee-api/bee-rest-api/src/types/responses.rs (hunk 4 above)
Comment (alexsporn) [reply bee-api/bee-rest-api/src/types/responses.rs:47]: You need to remove the serde rename

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 5]
```diff
@@ -58,6 +81,21 @@ pub struct ProtocolResponse {
     pub token_supply: String,
 }
 
+/// Returned in [`InfoResponse`].
+/// Rent information about the node.
```
Comment (thibault-martinez) [inline bee-api/bee-rest-api/src/types/responses.rs]: Not sure about that comment

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**